crates/blockifier/src/state/state_wrapper_test.rs

```diff
@@ -0,0 +1,199 @@
+use starknet_api::core::{ClassHash, CompiledClassHash, ContractAddress, Nonce};
+use starknet_api::hash::{StarkFelt, StarkHash};
+use starknet_api::state::StorageKey;
+
+use crate::state::cached_state::CachedState;
+use crate::state::state_api::{State, StateReader};
+use crate::state::state_wrapper::DynStateWrapper;
+
+#[test]
+fn set_class_hash_at() {
+    let contract_address = ContractAddress::from(1u128);
+    let mut state_1 = CachedState::default();
+
+    state_1.set_class_hash_at(contract_address, ClassHash(StarkHash::from(1u128))).unwrap();
+
+    let mut state_2 = DynStateWrapper::new(&mut state_1);
```

Comment:
nitpick: more descriptive names different than `state_1` and `state_2`. For example, `cached_state` and `wrapper_state`, not sure if those are the best